EBNF.Compiler/Compiler.cs

```diff
@@ -203,6 +203,39 @@ public static string[] CompileEbnf(string rawData, string namespaceName)
                         indentLevel++;
                         code.Add($"{new string('\t', 3 + indentLevel)}remainder = remainder.Substring({concatenation.Length - 2});");
                         break;
+                    case '!': // Regex Value
+                        
+                        code.Add($"var regex = new Regex({concatenation.Remove(0, 1)});");
+                        code.Add($"{new string('\t', 3 + indentLevel)}var match = regex.Match(remainder);");
+                        code.Add($"{new string('\t', 3 + indentLevel)}if (match.Success)");
+                        code.Add($"{new string('\t', 3 + indentLevel)}{{");
+                        indentLevel++;
+                        code.Add($"{new string('\t', 3 + indentLevel)}if (remainder.StartsWith(\" \"))");
```

Comment:
Please correct me if I'm wrong, but I believe this if statement could be combined with the one above by using the `&&` operator.

Anyway, please clarify what you are aiming to achieve with this test, as I feel that this condition is not the correct way to verify that a match has occurred precisely at the start of the String. (Remember Regex can match anywhere without the `^` or `$` operators.)